driver-legacy/src/main/com/mongodb/DBCollection.java

```diff
@@ -2447,10 +2447,11 @@ BulkWriteResult executeBulkWriteOperation(final boolean ordered, final Boolean b
         return retVal;
     }
 
-    DBObjectCodec getDefaultDBObjectCodec() {
-        return new DBObjectCodec(MongoClient.getDefaultCodecRegistry(),
+    Codec<DBObject> getDefaultDBObjectCodec() {
+        return new DBObjectCodec(getDB().getMongoClient().getCodecRegistry(),
```

Comment:
If we don't do this, then the uuidRepresentation won't get applied in this DBOBjectCodec

Reply:
Ack - is `MongoClient.getDefaultCodecRegistry` being deprecated or will it just default to the default UUID setting?

Reply:
It will default to UNSPECIFIED in 4.0, and will also have the uuidRepresentation from MongoClientOptions applied to it when it's used as the codecRegistry for MongoClientOptions